Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/LocalViewBookmarkLoader.cpp

```diff
@@ -0,0 +1,512 @@
+                // Once there is a UI for the view bookmarks these lines should be removed.
+                for (int i = 0; i < NumOfDefaultLocationsInLevel; i++)
+                {
+                    AZStd::string finalPath =
+                        "/" + m_bookmarkfileName + "/LocalBookmarks/" + AZStd::to_string(m_localBookmarkCount++);
+                    registry->SetObject(finalPath, ViewBookmark());
+                }
+
+                LoadViewBookmarks();
+                return true;
+            }
+            else
+            {
+                LoadViewBookmarks();
+
+                for (int i = 0; i < m_localBookmarks.size(); i++)
+                {
+                    AZStd::string finalPath = "/" + m_bookmarkfileName + "/LocalBookmarks/" + AZStd::to_string(i);
+                    registry->SetObject(finalPath, m_localBookmarks.at(i));
+                }
+
+                return true;
+            }
+        }
+        else // if the field is empty we don't have a file linked to the prefab, so we create one and we save it in the
+             // component
+        {
+            m_bookmarkfileName = GenerateBookmarkFileName();
+
+            // Initialize default locations to 0
+            for (int i = 0; i < NumOfDefaultLocationsInLevel; i++)
+            {
+                AZStd::string finalPath = "/" + m_bookmarkfileName + "/LocalBookmarks/" + AZStd::to_string(m_localBookmarkCount++);
+                registry->SetObject(finalPath, ViewBookmark());
+            }
+
+            LoadViewBookmarks();
+            bookmarkComponent->SetLocalBookmarksFileName(m_bookmarkfileName);
+            return true;
+        }
+    }
+
+    bool LocalViewBookmarkLoader::SaveLocalBookmark(const ViewBookmark& bookmark, ViewBookmarkType bookmarkType)
+    {
+        LoadDefaultLocalViewBookmarks();
+
+        AZStd::string finalPath;
+
+        switch (bookmarkType)
+        {
+        case ViewBookmarkType::Standard:
+            finalPath = "/" + m_bookmarkfileName + "/LocalBookmarks/" + AZStd::to_string(m_localBookmarkCount++);
+            break;
+        case ViewBookmarkType::LastKnownLocation:
+            finalPath = "/" + m_bookmarkfileName + "/LastKnownLocation";
+            break;
+        }
+
+        bool success = false;
+        // Write to the settings registry
+        if (auto registry = AZ::SettingsRegistry::Get())
+        {
+            success = registry->SetObject(finalPath, bookmark);
+        }
+
+        if (!success)
+        {
+             AZ_Warning(
+                 "LocalViewBookmarkLoader", false, "View Bookmark x=%.4f, y=%.4f, z=%.4f couldn't be saved", bookmark.m_position.GetX(),
+                 bookmark.m_position.GetY(), bookmark.m_position.GetZ());
+            return false; 
+        }
+
+        // If we managed to add the bookmark
+        SaveBookmarkSettingsFile();
+        LoadViewBookmarks();
```

Comment:
question: Do we need to call load here right after save? Might be worth adding a brief comment to explain _why_ this is needed